charts/Donut.args.js

```diff
@@ -0,0 +1,96 @@
+import React from "react";
+import Viz from "../primitives/Viz";
+
+import {Donut as D3plusDonut} from "d3plus-react";
+export const Donut = ({config}) => <D3plusDonut config={config} />;
+
+export const argTypes = {
+
+    /**
+     * Filters out unused argTypes from the Viz primitive and
+     * overrides any defaults that have been changed in Treemap
+     */
+    ...Object.keys(Viz.argTypes)
+      .filter(k => !k.match(/^(discrete|shape|zoom.*)$/))
+      .reduce((obj, k) => (obj[k] = Viz.argTypes[k], obj), {}),
```

Comment:
@davelandry let me know if this solution makes sense to you:

1. I imported the `Pie` argTypes into `Donut` using the following code renaming the Pie's argTypes as `pieArgTypes` because we are exporting the same variable in Donut argTypes definition:
```
import {argTypes as pieArgTypes} from "./Pie.args";
```
2. Changed `Viz.argTypes` to `pieArgTypes` and kept the `innerRadius` and `padPixel` specific methods in Donut argTypes definition:
```
export const argTypes = {

     /**
     * Filters out unused argTypes from the Pie args and
     * overrides any defaults that have been changed in Donut
     */
    ...Object.keys(pieArgTypes)
      .filter(k => !k.match(/^(discrete|shape|zoom.*)$/))
      .reduce((obj, k) => (obj[k] = pieArgTypes[k], obj), {}),
  
    /**
     * Donnut-specific methods
     */ 
    innerRadius: {...},
    padPixel: {...}
}
```

Reply:
This makes sense to me, but with 2 things:
1. make sure you still include the `Viz.argTypes` (the `pieArgTypes` can just be placed after it)
2. you can remove the `filter` for your `pieArgTypes`

Reply:
I'm not sure about the first one. If Pie uses `Viz.argTypes` as input and then we expand it using Pie args, is it still necessary to import Viz.argTypes if we are importing `pieArgTypes` into Donut args' file?
Regarding the second one, should I also remove the `filter` in Pie args' file?

Reply:
You are absolutely right about the first one, my bad!

But yes, I think you can still remove the filter, because that filter is for the Viz argTypes and is already happening in Pie.